portfolio/src/main/java/com/google/sps/servlets/NicknameSetter.java

```diff
@@ -0,0 +1,84 @@
+package com.google.sps.servlets;
+
+import com.google.sps.classes.Comment;
+import com.google.sps.classes.LoginHandler;
+import com.google.sps.classes.User;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+
+import javax.servlet.RequestDispatcher;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.swing.RepaintManager;
+
+@WebServlet("/set-nickname")
+public class NicknameSetter extends HttpServlet {
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    UserService userService = UserServiceFactory.getUserService();
+    String id = userService.getCurrentUser().getUserId();
+
+    Gson gson = new Gson();
+
+    Query q = new Query("Users").setFilter(new Query.FilterPredicate("id", Query.FilterOperator.EQUAL, id));
+    PreparedQuery result = datastore.prepare(q);
+    Entity givenEntity = result.asSingleEntity();
+
+    if (givenEntity == null){
```

Comment:
No, it is not. It is used to indicate that the user is not in the database and that is because they didn't set a nickname first time they logged in. After setting the nickname once, they will be added in the database. It is used to easily read in JS that the user doesn't have a nickname set.